coral/src/app/features/topics/request/TopicEditRequest.tsx

```diff
@@ -0,0 +1,270 @@
+import {
+  Alert,
+  Box,
+  Button,
+  Divider,
+  Input,
+  Option,
+  useToast,
+} from "@aivenio/aquarium";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import { SubmitHandler } from "react-hook-form";
+import { useNavigate, useParams, useSearchParams } from "react-router-dom";
+import { Dialog } from "src/app/components/Dialog";
+import {
+  Form,
+  NativeSelect,
+  SubmitButton,
+  TextInput,
+  Textarea,
+  useForm,
+} from "src/app/components/Form";
+import AdvancedConfiguration from "src/app/features/topics/request/components/AdvancedConfiguration";
+import SelectOrNumberInput from "src/app/features/topics/request/components/SelectOrNumberInput";
+import type { Schema } from "src/app/features/topics/request/form-schemas/topic-request-form";
+import formSchema from "src/app/features/topics/request/form-schemas/topic-request-form";
+import { generateTopicNameDescription } from "src/app/features/topics/request/utils";
+import { Environment } from "src/domain/environment";
+import { getAllEnvironmentsForTopicAndAcl } from "src/domain/environment/environment-api";
+import {
+  TopicDetailsPerEnv,
+  editTopic,
+  getTopicDetailsPerEnv,
+} from "src/domain/topic";
+import { HTTPError } from "src/services/api";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+function TopicEditRequest() {
+  const { topicName } = useParams();
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  const env = searchParams.get("env");
+
+  const [cancelDialogVisible, setCancelDialogVisible] = useState(false);
+
+  const { data: environments } = useQuery<Environment[], HTTPError>(
+    ["getEnvironmentsForTopicRequest"],
+    {
+      queryFn: () => getAllEnvironmentsForTopicAndAcl(),
+    }
+  );
+
+  const { data: topicDetailsForSourceEnv } = useQuery<
+    TopicDetailsPerEnv,
+    HTTPError
+  >(["getTopicDetailsPerEnv", topicName, env], {
+    queryFn: () =>
+      getTopicDetailsPerEnv({
+        topicname: topicName || "",
+        envSelected: env || "",
+      }),
+  });
+
+  const currentEnvironment = environments?.find(({ id }) => id === env);
```

Comment:
We need a handling for not finding a current Environment. A user could come through a link (or change the url param themself) or environments could be an empty list. Then the form just stays in a kind of loading state. 

I we need an error handling for `getTopicDetailsPerEnv` and `getEnvironmentsForTopicRequest`. If there's no response, the form stays either in loading or the user will get errors on invalid fields (but fields they didn't fill out / can't fill out). 

I think when this cases are handled, we can also get rid of the conditions in useEffect.

Reply:
Indeed :o

Reply:
Yes indeed, we should probably redirect to the Topic overview if env is not found.

However, we may not be able to remove the clause in the `useEffect`, as it's most likely the spot where we would do the redirecting?

Actually should probably have two `useEffect`s:
- one for error handling
- one for initializing form

Then we can have something like this, redirecting to topic overview when only the env is the issue, but redirecting to topic overview if topic name is the issue:

https://github.com/aiven/klaw/assets/20607294/ba3f8501-de76-417b-8709-b9f67c52812c

```

Reply:
Implemented here: [31ecf41](https://github.com/aiven/klaw/pull/1436/commits/31ecf410582b29fc94e0fdd32391cbbcf8b5145c)